### ISSUE TITLE
CNN convergence during training

### ISSUE THREAD

**Issue by ignaciorlando** [ignaciorlando/encoder-based-regularization#2]
1. Add a parameter with an epsilon to stop training.
2. Use the average validation loss in the last N epochs to stop training.
3. Use the number of epochs as the maximum number of epochs.
